interface/patient_file/pos_checkout_ippf.php

```diff
@@ -0,0 +1,2798 @@
+
+<style>
+    @media (min-width: 992px){
+        .modal-lg {
+            width: 1000px !Important;
+        }
+    }
+</style>
+
+<script>
+    var mypcc = <?php echo js_escape($GLOBALS['phone_country_code']); ?>;
+
+    <?php require($GLOBALS['srcdir'] . "/restoreSession.php"); ?>
+
+    // This clears tax amounts in preparation for recomputing taxes.
+    // TBD: Probably don't need this at all.
+    function clearTax(visible) {
+        var f = document.forms[0];
+        for (var i = 0; f['totaltax[' + i + ']']; ++i) {
+            f['totaltax[' + i + ']'].value = '0.00';
+        }
+    }
+
+    // This computes taxes and extended amount for the specified line, and returns
+    // the extended amount.
+    function calcTax(lino) {
+        var f = document.forms[0];
+        var pfx = 'line[' + lino + ']';
+        var taxable = parseFloat(f[pfx + '[charge]'].value);
+        var adjust = 0.00;
+        if (f[pfx + '[adjust]']) {
+            adjust = parseFloat(f[pfx + '[adjust]'].value);
+        }
+        var adjreason = '';
+        if (f[pfx + '[memo]']) {
+            adjreason = f[pfx + '[memo]'].value;
+        }
+        var extended = taxable - adjust;
+        if (true
+<?php
+    // Generate JavaScript that checks if the chosen adjustment type is to be
+    // applied before taxes are computed. option_value 1 indicates that the
+    // "After Taxes" checkbox is checked for an adjustment type.
+    $tmpres = sqlStatement(
+        "SELECT option_id FROM list_options WHERE " .
+        "list_id = 'adjreason' AND option_value = 1 AND activity = 1"
+    );
+    while ($tmprow = sqlFetchArray($tmpres)) {
+        echo "            && adjreason != '" . addslashes($tmprow['option_id']) . "'\n";
+    }
+    ?>
+        ) {
+            taxable -= adjust;
+        }
+        var taxnumrates  = f[pfx + '[taxnumrates]'].value;
+        var rates = taxnumrates.split(':');
+        for (var i = 0; i < rates.length; ++i) {
+            if (! f[pfx + '[tax][' + i + ']']) {
+                break;
+            }
+            var tax = 0;
+            if (f[pfx + '[billtime]'].value) {
+                // Line item is billed, use the tax amounts that were previously set for it.
+                tax = parseFloat(f[pfx + '[tax][' + i + ']'].value);
+            }
+            else {
+                tax = taxable * parseFloat(rates[i]);
+                tax = parseFloat(tax.toFixed(<?php echo $currdecimals ?>));
+                if (isNaN(tax)) {
+                    alert('Tax rate not numeric at line ' + lino);
+                }
+                f[pfx + '[tax][' + i + ']'].value = tax.toFixed(<?php echo $currdecimals ?>);
+            }
+            extended += tax;
+            var totaltax = parseFloat(f['totaltax[' + i + ']'].value) + tax;
+            f['totaltax[' + i + ']'].value = totaltax.toFixed(<?php echo $currdecimals ?>);
+        }
+        f[pfx + '[amount]'].value = extended.toFixed(<?php echo $currdecimals ?>);
```

Comment:
escape above $currdecimals with js_escape()

Reply:
I added code to force it to be an integer.